src/Hamiltonians/reduced_density_matrix.jl

```diff
@@ -206,8 +206,21 @@ function Interfaces.dot_from_right(
         ReducedDensityMatrixCalculcator!{TT,P}(left, dim),
         pairs(right)
     )
-    return hermitianpart!(ρ) # (ρ .+ ρ') ./ 2
+    return _check_diagonal!(ρ)
```

Comment:
```suggestion
    return ρ
```
I don't think that we have any right to expect that the diagonal matrix elements of the RDM be real if you take the dot product between arbitrary dvecs. Take the situation where you have two replicas that have acquired a global complex relative phase shift. The whole RDM will have that complex phase.